dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/RepositoryUtility.java

```diff
@@ -198,4 +205,14 @@ private static File findLocalRepository() {
     return managedDependencies;
   }
 
+  public static void registerMavenRepository(String mavenRepositoryUrl) {
```

Comment:
I'm having trouble following all the code in Github's poor excuse for a code review tool. However I am surprised that this method does not throw any checked exceptions.  What if the string is not a correct URL?

Reply:
It does not cause an exception when creating `RemoteRepository` instances.

When fetching dependencies, `DefaultTransporterProvider.newTransporter` searches for matching transporters (HttpTransporter and FileTransporter are there) for its protocol.

So I've added validation to check protocol to be `file`, `http` or `https`.